src/main/main.c

```diff
@@ -399,6 +445,17 @@ static int poll_init(struct re *re)
 
 	switch (re->method) {
 
+#ifdef HAVE_SELECT
+	case METHOD_SELECT:
+		if (re->fhsl)
+			return 0;
+
+		re->fhsl = mem_zalloc(re->maxfds * sizeof(re->fhsl), NULL);
```

Comment:
This gives a `Suspicious usage of 'sizeof(A*)'; pointer to aggregate` warning. We store only pointers, so I can use `sizeof(void *)` so this is more obvious.

Reply:
Ironically, that warning is supposed to prevent exactly the kind of errors like this - a missing dereference in `sizeof`. I would recommend you report this to the compiler devs.

Changing to `sizeof(struct re_fhs *)` or `sizeof(void *)` is fine.